game-core/src/main/java/games/strategy/engine/chat/PlayerChatRenderer.java

```diff
@@ -74,41 +89,16 @@ private String getNodeLabelWithPlayers(final PlayerName playerName) {
             : playerNames.stream().collect(Collectors.joining(", ", " (", ")")));
   }
 
-  private void setIconMap() {
-    final PlayerManager playerManager = game.getPlayerManager();
-    final PlayerList playerList;
+  private static PlayerList getPlayerList(final IGame game) {
     game.getData().acquireReadLock();
     try {
-      playerList = game.getData().getPlayerList();
+      return game.getData().getPlayerList();
     } finally {
       game.getData().releaseReadLock();
     }
-    // new HashSet removes duplicates
-    for (final INode playerNode : new HashSet<>(playerManager.getPlayerMapping().values())) {
-      final Set<String> players = playerManager.getPlayedBy(playerNode);
-      if (players.size() > 0) {
-        final List<Icon> icons =
-            players.stream()
-                .filter(player -> uiContext != null && uiContext.getFlagImageFactory() != null)
-                .map(
-                    player ->
-                        new ImageIcon(
-                            uiContext
-                                .getFlagImageFactory()
-                                .getSmallFlag(playerList.getPlayerId(player))))
-                .collect(Collectors.toList());
-        maxIconCounter = Math.max(maxIconCounter, icons.size());
-        playerMap.put(playerNode.getPlayerName().getValue(), players);
-        if (uiContext == null) {
-          iconMap.put(playerNode.getPlayerName().getValue(), null);
-        } else {
-          iconMap.put(playerNode.getPlayerName().getValue(), icons);
-        }
-      }
-    }
   }
 
-  public int getMaxIconCounter() {
-    return maxIconCounter;
```

Comment:
I double checked this is only called once per game instance, so this change is ok performance wise